rslib/src/import_export/mod.rs

```diff
@@ -1,10 +1,98 @@
 // Copyright: Ankitects Pty Ltd and contributors
 // License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
 
+mod gather;
+mod insert;
 pub mod package;
 
+use crate::prelude::*;
+
 #[derive(Debug, Clone, Copy, PartialEq)]
 pub enum ImportProgress {
-    Collection,
+    File,
     Media(usize),
+    MediaCheck(usize),
+    Notes(usize),
+}
+
+/// Wrapper around a progress function, usually passed by the [crate::backend::Backend],
+/// to make repeated calls more ergonomic.
+pub(crate) struct IncrementableProgress<P> {
+    progress_fn: Box<dyn FnMut(P, bool) -> bool>,
+    count_map: Option<Box<dyn FnMut(usize) -> P>>,
+    count: usize,
+}
+
+const UPDATE_INTERVAL: usize = 17;
```

Comment:
We could include this as an associated value, though not sure how much of an improvement that is.

```diff
diff --git a/rslib/src/import_export/mod.rs b/rslib/src/import_export/mod.rs
index 8f9825e59..b6afc04a4 100644
--- a/rslib/src/import_export/mod.rs
+++ b/rslib/src/import_export/mod.rs
@@ -23,9 +23,9 @@ pub(crate) struct IncrementableProgress<P> {
     count: usize,
 }

-const UPDATE_INTERVAL: usize = 17;
-
 impl<P> IncrementableProgress<P> {
+    const UPDATE_INTERVAL: usize = 17;
+
     /// `progress_fn: (progress, throttle) -> should_continue`
     pub(crate) fn new(progress_fn: impl 'static + FnMut(P, bool) -> bool) -> Self {
         Self {
@@ -47,7 +47,7 @@ impl<P> IncrementableProgress<P> {
     /// Must have called `set_count_map()` before calling this.
     pub(crate) fn increment(&mut self) -> Result<()> {
         self.count += 1;
-        if self.count % UPDATE_INTERVAL != 0 {
+        if self.count % Self::UPDATE_INTERVAL != 0 {
             return Ok(());
         }
         let progress = self.mapped_progress()?;
```

Alternatively we could optionally pass it into new. There may be a day that we have a routine where each iteration of the loop is long, and we don't want to wait 17 iterations to update.

Reply:
I made it a field of `Incrementor`. Should be easy enough to add a setter when the need arises. I think requesting it in `incrementor()` might be a bit irritating, because it would require consideration by the programmer each time, and in reality, we hardly have an idea what a reasonable value is, or do we?

Reply:
We could pass in Option<usize> with None meaning 17, but no problems keeping it as it is for now.